src/coreclr/vm/runtimehandles.cpp

```diff
@@ -1143,6 +1143,35 @@ FCIMPL2(FC_BOOL_RET, RuntimeTypeHandle::CompareCanonicalHandles, ReflectClassBas
 }
 FCIMPLEND
 
+FCIMPL1(Object*, RuntimeTypeHandle::GetThreadLocalAllocInternal, MethodTable* pMT)
+{
+    FCALL_CONTRACT;
+
+    _ASSERTE(pMT != nullptr);
+    _ASSERTE(pMT->IsValueType());
+    _ASSERTE(GCHeapUtilities::UseThreadAllocationContexts());
```

Comment:
I do not think that this is a valid assert. As discussed, there are configurations where UseThreadAllocationContexts is false.